plugin-niassa/src/main/java/ca/on/oicr/gsi/shesmu/pipedev/BaseProvenancePluginType.java

```diff
@@ -206,15 +207,192 @@ public ItemCache() {
       }
     }
 
+    private class SkippedItemCache
+        extends ValueCache<
+            Stream<CerberusFileProvenanceSkippedValue>,
+            Stream<CerberusFileProvenanceSkippedValue>> {
+      private SkippedItemCache() {
+        super(name + "-skipped " + fileName().toString(), 60, ReplacingRecord::new);
+      }
+
+      @Override
+      protected Stream<CerberusFileProvenanceSkippedValue> fetch(Instant lastUpdated) {
+        final AtomicInteger badFilePaths = new AtomicInteger();
+        final AtomicInteger badSets = new AtomicInteger();
+        final AtomicInteger badVersions = new AtomicInteger();
+        final Map<String, Integer> badSetCounts = new TreeMap<>();
+        return client
+            .map(BaseProvenancePluginType.this::fetch)
+            .orElseGet(Stream::empty)
+            .filter(
+                fp ->
+                    (fp.getSkip() == null || fp.getSkip().equals("true"))
```

Comment:
It looks like the standard CerberusFPValue also selects items where `fp.getSkip() == null`. Should those records be included in both CFPValue and CFPSkippedValue caches?

Reply:
They shouldn't get included in CFPSkippedValue as those records won't correspond with actually failed samples. I'll push an amendment.